*** a/include/SDL_events.h        2016-10-20 12:56:26.000000000 +0900
--- b/include/SDL_events.h        2021-02-11 03:58:32.954379999 +0900
*************** typedef struct SDL_KeyboardEvent
*** 192,197 ****
--- 192,198 ----
      Uint32 type;        /**< ::SDL_KEYDOWN or ::SDL_KEYUP */
      Uint32 timestamp;
      Uint32 windowID;    /**< The window with keyboard focus, if any */
+     Uint32 inputDevice; /**< webOS specific field */
      Uint8 state;        /**< ::SDL_PRESSED or ::SDL_RELEASED */
      Uint8 repeat;       /**< Non-zero if this is a key repeat */
      Uint8 padding2;