*** a/include/SDL_events.h        2016-10-20 12:56:26.000000000 +0900
--- b/include/SDL_events.h        2021-02-11 04:29:13.242063999 +0900
*************** typedef struct SDL_KeyboardEvent
*** 192,197 ****
--- 192,198 ----
      Uint32 type;        /**< ::SDL_KEYDOWN or ::SDL_KEYUP */
      Uint32 timestamp;
      Uint32 windowID;    /**< The window with keyboard focus, if any */
+     Uint32 inputSource; /**< webOS specific field */
      Uint8 state;        /**< ::SDL_PRESSED or ::SDL_RELEASED */
      Uint8 repeat;       /**< Non-zero if this is a key repeat */
      Uint8 padding2;
*************** typedef struct SDL_TextEditingEvent
*** 208,213 ****
--- 209,215 ----
      Uint32 type;                                /**< ::SDL_TEXTEDITING */
      Uint32 timestamp;
      Uint32 windowID;                            /**< The window with keyboard focus, if any */
+     Uint32 inputSource;                         /**< webOS specific field */
      char text[SDL_TEXTEDITINGEVENT_TEXT_SIZE];  /**< The editing text */
      Sint32 start;                               /**< The start cursor of selected editing text */
      Sint32 length;                              /**< The length of selected editing text */
*************** typedef struct SDL_TextInputEvent
*** 223,228 ****
--- 225,231 ----
      Uint32 type;                              /**< ::SDL_TEXTINPUT */
      Uint32 timestamp;
      Uint32 windowID;                          /**< The window with keyboard focus, if any */
+     Uint32 inputSource;                       /**< webOS specific field */
      char text[SDL_TEXTINPUTEVENT_TEXT_SIZE];  /**< The input text */
  } SDL_TextInputEvent;
  
